### PR TITLE
Apimanager secret reconciliation changes

### PR DESCRIPTION
**Pull request by miguelsorianod**
This PR makes secrets not being updated when no differences are detected. Differences are calculated from a set of specific fields that are taken into account

### CODE REVIEW

File: pkg/controller/apimanager/apimanager_controller.go [hunk 1]
```diff
@@ -151,111 +154,139 @@ func (r *ReconcileAPIManager) Reconcile(request reconcile.Request) (reconcile.Re
...
+	// We copy the secrets because we don't know the source of them. Might
+	// come from the Cache
+	currentCopy := current.DeepCopy()
+	desiredCopy := desired.DeepCopy()
+
+	// We convert StringData to Data because stringData cannot be read when
+	// obtained from the Kubernetes API and we need to compare the secret
+	// data
+	desiredCopy.Data = secretStringDataToData(desiredCopy.StringData)
+	if secretsEqual(currentCopy, desiredCopy) {
```
Comment (eguzki) [inline pkg/controller/apimanager/apimanager_controller.go]: `updateSecret` method name makes me think you are updating secrets. Why not check if secrets are equal and if not call `updateSecret`?

**Comment by eguzki** [comment]
In general, I looks ok to me. 

On comment though, try to add changes that stick to the PR purpose. For example, try not to add changes that do not have to do with secret reconciliation, like new struct attributes of `ReconcileAPIManager`. It is ok for now.

File: pkg/controller/apimanager/apimanager_controller.go (hunk 1 above)
Comment (miguelsorianod) [reply pkg/controller/apimanager/apimanager_controller.go]: That's what I'm doing later in https://github.com/3scale/3scale-operator/blob/834e7b6b050731578d08bbda4fd443c954f39168/pkg/controller/apimanager/apimanager_controller.go#L211

Comment (miguelsorianod) [reply pkg/controller/apimanager/apimanager_controller.go]: Maybe renaming the "updateSecret" method to "reconcileSecret" ?

Comment (miguelsorianod) [reply pkg/controller/apimanager/apimanager_controller.go]: I renamed the method updateSecret to reconcileSecret to make more clear that not always an update will be produced

**Pull request merged**